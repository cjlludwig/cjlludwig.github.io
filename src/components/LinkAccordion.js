import { useState } from "react";
import Accordion from 'react-bootstrap/Accordion';
import { ReactTinyLink } from 'react-tiny-link';
import linkPic from '../static/images/web-link.png';


import 'bootstrap/dist/css/bootstrap.min.css';

function LinkAccordion({ links=[] }) {
  const [ openKey, setOpen ] = useState(null);
  // const proxy = process.env.REACT_APP_ENV === "dev" ? "https://proxy.cors.sh" : "";

  return (
    <Accordion >
    {
      links.map((data, index) => {
        const { name, url, description } = data;
        return (
          <Accordion.Item key={index} eventKey={index + 1}>
            <Accordion.Header
              onClick={() => setOpen(index)}
            >
             {name}
            </Accordion.Header>
            <Accordion.Body>
              {openKey === index &&
                <ReactTinyLink
                  width="100%"
                  description={description}
                  defaultMedia={linkPic}
                  cardSize="small"
                  showGraphic={true}
                  maxLine={3}
                  minLine={2}
                  url={url}
                  proxyUrl={""}
                  requestHeaders={ { 
                    "Access-Control-Allow-Origin": "*" ,
                    "Access-Control-Allow-Headers":"*",
                    'Access-Control-Allow-Methods': 'HEAD, GET, OPTIONS',
                    "Access-Control-Max-Age": 1000,
                  } }
                />
              } 
            </Accordion.Body>
          </Accordion.Item>
        )
      })
    }
  </Accordion>
  );
}

export default LinkAccordion;
