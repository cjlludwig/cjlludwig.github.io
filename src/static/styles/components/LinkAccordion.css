.link-pic {
  height: 20vmin;
  padding: 10px;
}

.link-span {
  display: inline-flex;
  width: 100%;
  align-items: center;
}

.link-card {
  width: 100%;
}