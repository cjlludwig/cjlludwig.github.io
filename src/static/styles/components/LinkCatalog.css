.link-catalog-wrapper {
  padding-top: 3%;
  padding-bottom: 7%;
  padding-right: 5%;
  padding-left: 5%;
}

.header {
  width: 120px;
}
.link-header {
  color: white;
  text-align: center;
  padding-bottom: 10px;
}

.description {
  padding-left: 1.5%
}
.title-header {
  float: left;
  width: 15%;
  min-width: 120px;
  text-align: center;
  font-size: 1.25rem;
}
.header-description {
  float: left;
  padding-left: 1.5%;
  width: 75%;
  text-align: center;
  font-size: 1.25rem;

}

.accordion-button {
  font-weight: bold;

}

/* Mobile */
@media screen and (max-width:960px) {
  /* CSS for screens that are 480px pixels or less will be put in this section */
  .title-header {
    width: 90px;
    min-width: 90px;
    font-size: 1.1rem;
  }
  .header-description {
    width: 65%;
    font-size: 1.1rem;
  }
}


/* React */
/* .accordion {
  --bs-accordion-color: #212529;
  --bs-accordion-bg: #fff;
  --bs-accordion-transition: color 0.15s ease-in-out,background-color 0.15s ease-in-out,border-color 0.15s ease-in-out,box-shadow 0.15s ease-in-out,border-radius 0.15s ease;
  --bs-accordion-border-color: var(--bs-border-color);
  --bs-accordion-border-width: 1px;
  --bs-accordion-border-radius: 0.375rem;
  --bs-accordion-inner-border-radius: calc(0.375rem - 1px);
  --bs-accordion-btn-padding-x: 1.25rem;
  --bs-accordion-btn-padding-y: 1rem;
  --bs-accordion-btn-color: #212529;
  --bs-accordion-btn-bg: var(--bs-accordion-bg);
  --bs-accordion-btn-icon: url(data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 viewBox=%270 0 16 16%27 fill=%27%23212529%27%3e%3cpath fill-rule=%27evenodd%27 d=%27M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z%27/%3e%3c/svg%3e);
  --bs-accordion-btn-icon-width: 1.25rem;
  --bs-accordion-btn-icon-transform: rotate(-180deg);
  --bs-accordion-btn-icon-transition: transform 0.2s ease-in-out;
  --bs-accordion-btn-active-icon: url(data:image/svg+xml,%3csvg xmlns=%27http://www.w3.org/2000/svg%27 viewBox=%270 0 16 16%27 fill=%27%230c63e4%27%3e%3cpath fill-rule=%27evenodd%27 d=%27M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z%27/%3e%3c/svg%3e);
  --bs-accordion-btn-focus-border-color: #86b7fe;
  --bs-accordion-btn-focus-box-shadow: 0 0 0 0.25rem rgba(13, 110, 253, 0.25);
  --bs-accordion-body-padding-x: 1.25rem;
  --bs-accordion-body-padding-y: 1rem;
  --bs-accordion-active-color: #0c63e4;
  --bs-accordion-active-bg: #e7f1ff;
} */