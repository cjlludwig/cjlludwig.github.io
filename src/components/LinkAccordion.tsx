import { useState } from "react";
import Accordion from 'react-bootstrap/Accordion';
import { Links } from '../models/links';
import linkPic from '../static/images/web-link.png';
import { Card } from "react-bootstrap";
import '../static/styles/components/LinkAccordion.css';
import 'bootstrap/dist/css/bootstrap.min.css';

function LinkAccordion<Props extends { links: Links[] }>(props: Props) {
  const { links=[] } = props;
  const [ openKey, setOpen ] = useState<number | null>(null);

  return (
    <Accordion >
    {
      links.map((data, index) => {
        const { name, url, description } = data;
        return (
          <Accordion.Item key={index} eventKey={`${index+1}`}>
            <Accordion.Header
              onClick={() => setOpen(index)}
            >
             {name}
            </Accordion.Header>
            <Accordion.Body>
              {openKey === index &&
              <a href={url} style={{ textDecoration: "none", color: "inherit" }}>
                <span className="link-span">
                  <img src={linkPic} className="link-pic" alt="default link" />
                  <Card className="link-card">
                    <Card.Header>{name}</Card.Header>
                    <Card.Body>{description}</Card.Body>
                    <Card.Footer style={{ textDecoration: "underline", color: "blue" }}>{url}</Card.Footer>
                  </Card>
                </span>
              </a>
              } 
            </Accordion.Body>
          </Accordion.Item>
        )
      })
    }
  </Accordion>
  );
}

export default LinkAccordion;
