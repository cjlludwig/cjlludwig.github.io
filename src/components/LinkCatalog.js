import '../static/styles/components/LinkCatalog.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import LinkAccordion from './LinkAccordion';

const articles = [
  {
    name: "Politics of ChatGPT",
    description: "Great article done on the political sentiment of the AI tool ChatGPT. The author used some standard political sentiment quizzes to check the AI's alignment and offered great visualizations to demonstrate it's change over time.",
    url: "https://davidrozado.substack.com/p/chatgpt"
  },
  {
    name: "Tao of Node",
    description: "Opinionated piece with some interesting and useful insights on Node Best Practices.",
    url: "https://alexkondov.com/tao-of-node/"
  },
  {
    name: "2022 Most Popular Node Frameworks",
    description: "Overview of the most commonly used frameworks in Node apps pulled from a variety of dev surveys.",
    url: "https://stackdiary.com/node-js-frameworks/"
  },
  {
    name: "Dr. Werner Vogels Keynote",
    description: "Dr. Werner Vogels AWS Re:Invent kyenote covering interesting architecture patterns and technologies.",
    url: "https://www.youtube.com/watch?v=RfvL_423a-I"
  },
];

const tools = [
  {
    name: "Vs Code Dev Containers",
    description: "An awesome new extension in Vs Code that allows developers to associate dvelopment containers with their code. Greatly reduces start up costs for new team members.",
    url: "https://code.visualstudio.com/docs/devcontainers/containers"
  },
  {
    name: "react-gh-pages",
    description: "An easy to use framework for deploying React apps to GitHub pages.",
    url: "https://github.com/gitname/react-gh-pages"
  },
  {
    name: "Coolors",
    description: "Color palette generator for websites.",
    url: "https://coolors.co/"
  },
  {
    name: "Purecss",
    description: "Simple, compact CSS modules for use in new projects.",
    url: "https://purecss.io/"
  },
  {
    name: "Docker-OSX",
    description: "A containerized version of Mac OSX.",
    url: "https://github.com/sickcodes/Docker-OSX"
  }
];

const news = [
  {
    name: "Node Weekly Newsletter",
    description: "Newsletter I use to keep up with Node framework changes.",
    url: "https://nodeweekly.com/"
  },
  {
    name: "React Weekly Newsletter",
    description: "Newsletter I use to keep up with React framework changes.",
    url: "https://react.statuscode.com/"
  },
  {
    name: "TLDR Weekly Newsletter",
    description: "Software engineering newsletter I use to keep up with the indurstry.",
    url: "https://tldr.tech/"
  },
  {
    name: "O'Reilly Radar Trends to Watch Newsletter",
    description: "Software engineering newsletter I use to keep up with the indurstry.",
    url: "https://www.oreilly.com/emails/newsletters/"
  },
  {
    name: "Programming Subreddit",
    description: "Forum for developers to post interesting topics and articles.",
    url: "https://www.reddit.com/r/programming/"
  },
  {
    name: "Programming Humor Subreddit",
    description: "Forum for developers to post the latest and greatest memes and technology jokes. Great for a good laugh!",
    url: "https://www.reddit.com/r/ProgrammerHumor/"
  }
]

function LinkCatalog() {
  return (
    <div className='link-catalog-wrapper'>
      <h3 className='link-header'>
        Favorite Tools
      </h3>
      <LinkAccordion links={tools}/>
      <h3 style={{ paddingTop: "15px" }} className='link-header'>
        Tech News Sources
      </h3>
      <LinkAccordion links={news}/>
      <h3 style={{ paddingTop: "15px" }} className='link-header'>
        Interesting Articles
      </h3>
      <LinkAccordion links={articles}/>
    </div>
  );
}

export default LinkCatalog; 