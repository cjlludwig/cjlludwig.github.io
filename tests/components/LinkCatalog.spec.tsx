import React from "react";
import renderer from "react-test-renderer";

import LinkCatalog from "../../src/components/LinkCatalog";
import LinkAccordion from "../../src/components/LinkAccordion";

jest.mock("../../src/components/LinkAccordion");

describe("<LinkCatalog />", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<LinkCatalog />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});