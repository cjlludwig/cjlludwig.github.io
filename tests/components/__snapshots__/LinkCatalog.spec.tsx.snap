// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkCatalog /> renders correctly 1`] = `
<div
  className="link-catalog-wrapper"
>
  <h3
    className="link-header"
  >
    Favorite Tools
  </h3>
  <h3
    className="link-header"
    style={
      {
        "paddingTop": "15px",
      }
    }
  >
    Tech News Sources
  </h3>
  <h3
    className="link-header"
    style={
      {
        "paddingTop": "15px",
      }
    }
  >
    Interesting Articles
  </h3>
</div>
`;
